### PR TITLE
Add "simulate" command.

### PR DESCRIPTION
**Pull request by ZehMatt**
Before I forget about it again. 
This a basic command to simulate the game for a certain amount of ticks resulting the same network hash we use to identify desyncs. I used this to compare game logic across multiple builds and perhaps someone could actually put that to the CI to see if a PR will actually change the game logic and warn us about it.

### CODE REVIEW

File: src/openrct2/cmdline/SimulateCommands.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
+#pragma region Copyright(c) 2014 - 2017 OpenRCT2 Developers
```
Comment (janisozaur) [inline src/openrct2/cmdline/SimulateCommands.cpp]: 2018

File: src/openrct2/cmdline/SimulateCommands.cpp [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    gOpenRCT2Headless = true;
+
+#ifndef DISABLE_NETWORK
+    gNetworkStart = NETWORK_MODE_SERVER;
+#endif
+
+    std::unique_ptr<IContext> context(CreateContext());
+    if (context->Initialise())
+    {
+        context->InitialiseDrawingEngine();
```
Comment (janisozaur) [inline src/openrct2/cmdline/SimulateCommands.cpp]: Is it needed? Or is it a leftover from a headful version?

File: src/openrct2/cmdline/SimulateCommands.cpp [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        {
+            context->GetGameState()->UpdateLogic();
+        }
+        Console::WriteLine("Completed: %s", sprite_checksum());
+
+        context->DisposeDrawingEngine();
+    }
+    else
+    {
+        Console::Error::WriteLine("Context initialization failed.");
```
Comment (janisozaur) [inline src/openrct2/cmdline/SimulateCommands.cpp]: Should exit with error

File: src/openrct2/cmdline/SimulateCommands.cpp (hunk 1 above)
Comment (ZehMatt) [reply src/openrct2/cmdline/SimulateCommands.cpp]: Ops, I'll copy the more recent copyright notice.

**Pull request merged**